operator-api/src/xline.rs

```diff
@@ -0,0 +1,147 @@
+use async_trait::async_trait;
+use k8s_openapi::api::core::v1::Pod;
+use kube::api::{AttachParams, AttachedProcess};
+use kube::Api;
+use std::fmt::{Debug, Formatter};
+
+use std::process::{Child, Command};
+
+/// xline handle abstraction
+#[async_trait]
+pub trait XlineHandle: Debug + Send + Sync + 'static {
+    /// start a xline node
+    async fn start(&mut self) -> anyhow::Result<()>; // we dont care about what failure happened when start, it just failed
+
+    /// kill a xline node
+    async fn kill(&mut self) -> anyhow::Result<()>;
+}
+
+/// Local xline handle, it will execute the xline in the local
+/// machine with the start_cmd
+#[derive(Debug)]
+pub struct LocalXlineHandle {
+    start_cmd: String,
+    child_proc: Option<Child>,
+}
+
+impl LocalXlineHandle {
+    /// New a local xline handle
+    pub fn new(start_cmd: String) -> Self {
+        Self {
+            start_cmd,
+            child_proc: None,
+        }
+    }
+}
+
+#[async_trait]
+impl XlineHandle for LocalXlineHandle {
+    async fn start(&mut self) -> anyhow::Result<()> {
+        self.kill().await?;
+        let mut cmds = self.start_cmd.split_whitespace();
```

Comment:
Could we use Vec<String> as the type of `start_cmd` directly?

Reply:
It is refactored in the next PRs